pkg/clusteragent/languagedetection/telemetry.go

```diff
@@ -24,4 +29,140 @@ var (
 		"Tracks the number of patch requests sent by the patcher to the kubernetes api server",
 		commonOpts,
 	)
+
+	queueDepth = telemetry.NewGaugeWithOpts(
```

Comment:
Regarding the queue telemetry, I think we can reuse the code used for the SBOM telemetry (maybe move it to its own package). 
The argument of the `New*Metric(string)` functions can be the subsystem so for example we can use `NewDepthMetric("sbom")` for sbom and `NewDepthMetric("pld_patcher")` for process language detection.